crates/ln-dlc-node/src/node/mod.rs

```diff
@@ -382,12 +388,12 @@ impl Node {
                 // TODO: Check why we need to announce the node of the mobile app as otherwise the
                 // just-in-time channel creation will fail with a `unable to decode own hop data`
                 // error.
-
-                let address = util::build_net_address(address.ip(), address.port());
                 let mut interval = tokio::time::interval(BROADCAST_NODE_ANNOUNCEMENT_INTERVAL);
 
                 loop {
-                    peer_manager.broadcast_node_announcement([0; 3], alias, vec![address.clone()]);
+                    tracing::debug!("Announcing node on {:?}", announcements);
+                    let announcements = announcements.clone();
+                    peer_manager.broadcast_node_announcement([0; 3], alias, announcements);
```

Comment:
I think this loop should actually be removed and we only announce once.

Reply:
Well, I thought about it, but left it for the following reasons.

1. We use the same code for our mobile node and there it is useful to announce the nodes ip address in a loop. (E.g. the network changes)
2. In case we want to use a domain instead of an ip to announce our node, we could a) change the ip underneath without having to change the config on the coordinator and b) also work with dyndns.

Reply:
> We use the same code for our mobile node and there it is useful to announce the nodes ip address in a loop. (E.g. the network changes)

Do we need to announce the app node? 

> In case we want to use a domain instead of an ip to announce our node, we could a) change the ip underneath without having to change the config on the coordinator and b) also work with dyndns.

I thought LN does not work with domains?

Reply:
> Do we need to announce the app node?

It looks like it. see also 

https://github.com/get10101/10101/blob/a045dc942e22282a90b89cba9b195a0ca9e21194/crates/ln-dlc-node/src/node/mod.rs#L380-L382

But it shouldn't be required, I guess?

> I thought LN does not work with domains?

Well LN Bolt-07 supports it

```
The following address descriptor types are defined:
...
    5: DNS hostname; data = [1:hostname_len][hostname_len:hostname][2:port] (length up to 258)
        hostname bytes MUST be ASCII characters.
        Non-ASCII characters MUST be encoded using Punycode: https://en.wikipedia.org/wiki/Punycode
```
https://github.com/lightning/bolts/blob/master/07-routing-gossip.md

It's just not supported by LND yet - https://github.com/lightningnetwork/lnd/issues/6337

However, we don't have to announce on the hostname, we could still do the announcement on ip and resolve the dns ourselves.